### PR TITLE
[PVR] Speed up persisting EPG container to database

### LINKED ISSUE #17800: Long shutdown time and intensive HDD usage updating EPG for large IPTV playlists
### Describe the bug

Kodi creates and deletes the `Epg*.db-journal` for every EPG item being persisted. This causes excessive HDD wear and long shutdown times (2-5 minutes) if IPTV playlist and EPG have several thousands of live TV channels and VODs.

### Expected Behavior

The journal files for `Epg*.db` should be processed in memory so shutdown times are instant (~5-8 seconds on the same playlist / EPG pair vs the unmodified setup)

### Actual Behavior

The iotop graph is shown below:

iotop-excessive-hdd-usage

The shutdown time is very long (2-5 minutes on AMD E-300 dual-core CPU and 5400rpm HDD)

### Possible Fix

I created the script and the systemd unit mounting the `$KODIPROFILE/userdata/Database` as an overlayfs with the lower directory of original `Database` directory and `tmpfs` mount as upper-directory:

[kodi-mount-overlay.sh](https://github.com/xbmc/xbmc/files/4573561/kodi-mount-overlay.sh.txt)
[kodi-mount-overlay.service](https://github.com/xbmc/xbmc/files/4573566/kodi-mount-overlay.service.txt)

With this ugly hack, the shutdown time and disk usage is desirable.

### To Reproduce

1. Generate the test playlist / EPG pair using the script:
[generate-big-test-m3u-epg.sh](https://github.com/xbmc/xbmc/files/4573573/generate-big-test-m3u-epg.sh.txt)

2. Load the M3U and EPG into IPTV Simple Client and let Kodi process the channels and EPG markers
3. Exit Kodi and observe high disk usage running `sudio iotop` and lots of created-and-vanishing `Epg36.db-journal` in `$KODIPROFILE/userdata/Database`

### Debuglog

The debuglog can be found here:

[kodi.log](https://github.com/xbmc/xbmc/files/4573629/kodi.log)

The log ended after I had to issue
``` shell
kill -9 $(pidof kodi-x11)
```

on my test VM.

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [x] Linux
 - [ ] OSX
 - [ ] Raspberry-Pi
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: Debian 10 buster amd64
 - Kodi version: 19.0 trunk

### PR DESCRIPTION
**Pull request by ksooo**
Fixes #17800 

Runtime-tested by me and at least two users on latest Kodi master.

@phunkyfish when you have some time for a review/test.

### CODE REVIEW

**Review by phunkyfish — APPROVED** [review]
Nice ;)

File: xbmc/pvr/epg/EpgDatabase.cpp [hunk 1]
```diff
@@ -958,7 +958,7 @@ bool CPVREpgDatabase::GetLastEpgScanTime(int iEpgId, CDateTime* lastScan)
   return bReturn;
 }
 
-bool CPVREpgDatabase::PersistLastEpgScanTime(int iEpgId, const CDateTime& lastScanTime, bool bQueueWrite /* = false */)
+bool CPVREpgDatabase::PersistLastEpgScanTime(int iEpgId, const CDateTime& lastScanTime, bool bQueueWrite)
```
Comment (Rechi) [inline xbmc/pvr/epg/EpgDatabase.cpp:961]: > The `ColumnLimit` in `.clang-format` is set to `100` which defines line length [...]

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#3-formatting

**Review by Rechi — COMMENTED** [review]
Please follow the [current code guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md).

File: xbmc/pvr/epg/EpgDatabase.cpp (hunk 1 above)
Comment (ksooo) [reply xbmc/pvr/epg/EpgDatabase.cpp]: Fixed

**Comment by garbear** [comment]
To satisfy my thirst for numbers, do you have stats, rough estimates or general feelings on the improvement?

**Comment by basilgello** [comment]
> To satisfy my thirst for numbers, do you have stats, rough estimates or general feelings on the improvement?

I wrote in the issue, that the patch reduced Kodi shutdown time to 2s from ~1min. You can grab the reproducer on a clean installation and test yourself with unmodified version first :)

**Comment by ksooo** [comment]
@garbear more numbers given in #17800

**Pull request merged**